contracts/protocol/modules/v1/AaveV3LeverageModule.sol

```diff
@@ -0,0 +1,1151 @@
+     * @param _totalRedeemAmount    Total amount of `_collateralAsset` being sold
+     * @param _totalRepayAmount     Total amount of `_repayAsset` being repaid
+     * @param _protocolFee          Protocol fee charged
+     */
+    event LeverageDecreased(
+        ISetToken indexed _setToken,
+        IERC20 indexed _collateralAsset,
+        IERC20 indexed _repayAsset,
+        IExchangeAdapter _exchangeAdapter,
+        uint256 _totalRedeemAmount,
+        uint256 _totalRepayAmount,
+        uint256 _protocolFee
+    );
+
+    /**
+     * @dev Emitted on addCollateralAssets() and removeCollateralAssets()
+     * @param _setToken Instance of SetToken whose collateral assets is updated
+     * @param _added    true if assets are added false if removed
+     * @param _assets   Array of collateral assets being added/removed
+     */
+    event CollateralAssetsUpdated(
+        ISetToken indexed _setToken,
+        bool indexed _added,
+        IERC20[] _assets
+    );
+
+    /**
+     * @dev Emitted on addBorrowAssets() and removeBorrowAssets()
+     * @param _setToken Instance of SetToken whose borrow assets is updated
+     * @param _added    true if assets are added false if removed
+     * @param _assets   Array of borrow assets being added/removed
+     */
+    event BorrowAssetsUpdated(
+        ISetToken indexed _setToken,
+        bool indexed _added,
+        IERC20[] _assets
+    );
+
+    /**
+     * @dev Emitted when `underlyingToReserveTokensMappings` is updated
+     * @param _underlying           Address of the underlying asset
+     * @param _aToken               Updated aave reserve aToken
+     * @param _variableDebtToken    Updated aave reserve variable debt token
+     */
+    event ReserveTokensUpdated(
+        IERC20 indexed _underlying,
+        IAToken indexed _aToken,
+        IVariableDebtToken indexed _variableDebtToken
+    );
+
+    /**
+     * @dev Emitted on updateAllowedSetToken()
+     * @param _setToken SetToken being whose allowance to initialize this module is being updated
+     * @param _added    true if added false if removed
+     */
+    event SetTokenStatusUpdated(
+        ISetToken indexed _setToken,
+        bool indexed _added
+    );
+
+    /**
+     * @dev Emitted on updateAnySetAllowed()
+     * @param _anySetAllowed    true if any set is allowed to initialize this module, false otherwise
+     */
+    event AnySetAllowedUpdated(
+        bool indexed _anySetAllowed
+    );
+
+    /* ============ Constants ============ */
+
+    // This module only supports borrowing in variable rate mode from Aave which is represented by 2
+    uint256 constant internal BORROW_RATE_MODE = 2;
+
+    // String identifying the DebtIssuanceModule in the IntegrationRegistry. Note: Governance must add DefaultIssuanceModule as
+    // the string as the integration name
+    string constant internal DEFAULT_ISSUANCE_MODULE_NAME = "DefaultIssuanceModule";
```

Comment:
Is that what we did on Index-Protocol?

Reply:
What exactly do you mean ? 
Yes we will have to register the debtIssuanceModule as the "DefaultIssuanceModule" integration on the index-protocol integrationRegistry. We can only do this after we deploy the aaveV3LeverageModule though. 

See test for example:
https://github.com/IndexCoop/index-protocol/pull/17/files#diff-7c940931b5f46f7ede0a4c22b5773c41e0c01dd018f89a25debb22fc5fa41dc7R212


Does that answer the question ?

Reply:
Since we haven't deployed the aaveLeverageModule on index-protocol we did not have to do this yet I think. (Although it must have been done by set when deploying the original ALM).

Reply:
Ah I see - yeah sgtm